kythe/extractors/openjdk11/java-wrapper.go

```diff
@@ -0,0 +1,230 @@
+	defaultJdkVersion = "11"
+	extractorJarPath  = "io_kythe/kythe/java/com/google/devtools/kythe/extractors/java/standalone/javac9_extractor_deploy.jar"
+)
+
+var (
+	modulePattern = regexp.MustCompile("^@/.*/_the.(.*)_batch.tmp$")
+)
+
+func moduleName() string {
+	// Hackish way to determine the likely module being compiled.
+	return modulePattern.ReplaceAllString(os.Args[len(os.Args)-1], "$1")
+}
+
+func outputDir() string {
+	val := os.Getenv(kytheOutputVar)
+	if len(val) == 0 {
+		usr, err := user.Current()
+		if err != nil {
+			log.Fatalf("ERROR: unable to determine current user: %v", err)
+		}
+		val = filepath.Join(usr.HomeDir, "kythe-openjdk11-output")
+	}
+	return val
+}
+
+func targetJdk() string {
+	if val := os.Getenv(jdkVersionVar); len(val) > 0 {
+		return val
+	}
+	return defaultJdkVersion
+}
+
+func checkPath(parts ...string) error {
+	_, err := os.Stat(filepath.Join(parts...))
+	return err
+}
+
+func setupRunfiles() error {
+	if len(os.Getenv("RUNFILES_DIR")) > 0 || len(os.Getenv("RUNFILES_MANIFEST_FILE")) > 0 {
+		return nil
+	}
+	for _, base := range []string{os.Args[0] + ".runfiles", "."} {
+		if root, err := filepath.Abs(base); err == nil {
+			if _, err := os.Stat(root); err == nil {
+				os.Setenv("RUNFILES_DIR", root)
+				if _, err := os.Stat(filepath.Join(root, "MANIFEST")); err == nil {
+					os.Setenv("RUNFILES_MANIFEST_FILE", filepath.Join(root, "MANIFEST"))
+				}
+				return nil
+			}
+		}
+	}
+	return errors.New("unable to setup runfiles")
+}
+
+func findJavaCommand() (string, error) {
+	if val := os.Getenv(javaCommandVar); len(val) > 0 {
+		return val, checkPath(val)
+	}
+	if val := os.Getenv(javaHomeVar); len(val) > 0 {
+		return filepath.Join(val, "bin", "java"), checkPath(val, "bin", "java")
+	}
+	return exec.LookPath("java")
+}
+
+func findExtractorJar() (string, error) {
+	if val := os.Getenv(extractorJarVar); len(val) > 0 {
+		return val, checkPath(val)
+	}
+	return bazel.Runfile(extractorJarPath)
+}
+
+func extractorArgs(args []string, jar string) []string {
+	target := targetJdk()
+	isJavac := false
+	isBootstrap := true
+	var result []string
+	for len(args) > 0 {
+		switch a := shift(&args); a {
+		case "-m", "--module":
+			v := shift(&args)
+			if !strings.HasSuffix(v, ".javac.Main") {
+				// This is not a javac invocation, don't extract.
+				return nil
+			}
+			isJavac = true
+			result = append(result,
+				"--add-exports=jdk.compiler.interim/com.sun.tools.javac.main=ALL-UNNAMED",
+				"--add-exports=jdk.compiler.interim/com.sun.tools.javac.util=ALL-UNNAMED",
+				"--add-exports=jdk.compiler.interim/com.sun.tools.javac.file=ALL-UNNAMED",
+				"--add-exports=jdk.compiler.interim/com.sun.tools.javac.api=ALL-UNNAMED",
+				"--add-exports=jdk.compiler.interim/com.sun.tools.javac.code=ALL-UNNAMED",
+				"-jar", jar, "-Xprefer:source")
+		case "-target":
+			v := shift(&args)
+			if v != target {
+				// This is definitely a bootstrap compilation, don't extract it.
```

Comment:
Little more explanation why?

Reply:
I figured there's no harm in extraction the bootstrap compilations as well, so removed this and related code.

Reply:
(It doesn't take appreciably longer and should give more complete coverage of the repository itself, even if that's not strictly necessary for our purposes).